apis/v1alpha2/httproute_types.go

```diff
@@ -219,7 +219,7 @@ type HTTPRouteRule struct {
 	// MUST receive a 500 status code.
 	//
 	// When a BackendRef refers to a Service that has no ready endpoints, it is
-	// recommended to return a 503 status code.
+	// recommended to return a 500 status code.
```

Comment:
This is a tricky one. I think 503 could actually be appropriate here. In this specific case the Service reference is correct, there just aren't any endpoints available at that point in time. That seems more likely to be the kind of temporary outage suggested by a 503 than any of the other cases we've considered (ie a reference to a resource that simply doesn't exist).

With that said, a 500 is more generic and consistent with what we've used elsewhere. I think I'm good with this change, but want a few more people to chime in to confirm.

Reply:
Agree 503 could actually make sense here, I had assumed it was inconsistent, but it's more just splitting out "valid backendRef with zero instances" as a separate case from "invalid backendRef pointing to nonexistant thing"

Reply:
Do we keep 503 then?

Reply:
I think we should keep 503 for this specific case, only.